src/Compilers/CSharp/Portable/Symbols/Source/SourceEventSymbol.cs

```diff
@@ -30,6 +30,7 @@ internal abstract class SourceEventSymbol : EventSymbol, IAttributeTargetSymbol
         private SymbolCompletionState _state;
         private CustomAttributesBag<CSharpAttributeData> _lazyCustomAttributesBag;
         private string _lazyDocComment;
+        private string _lazyExpandedDocComment;
```

Comment:
>private string _lazyExpandedDocComment; [](start = 8, length = 39)

Is there a big advantage to cache this? Is it common to request this information?

Reply:
I'm not sure how valuable the cache is. I have observed documentation comment evaluation as a substantial contributor to some poor performance scenarios, but that's not a direct indicator of the helpfulness of the cache (in theory or as implemented). This commit was part of an experimental branch where I updated Quick Info to expand included documentation as part of the presentation process (which I believe is the correct approach), only to find that in some cases the results were not getting expanded.

Reply:
> I'm not sure how valuable the cache is.

We might want to take approach used for PE Symbols. It looks like implementations of GetDocumentationCommentXml for them cache result of the first request and recalculate if subsequent requests use different parameters.

---
In reply to: [288692002](https://github.com/dotnet/roslyn/pull/36021#discussion_r288692002) [](ancestors = 288692002)

Reply:
> We might want to take approach used for PE Symbols. It looks like implementations of GetDocumentationCommentXml for them cache result of the first request and recalculate if subsequent requests use different parameters.

The PE symbol implementation looks at the culture info, which (IMO) seems less likely to change during a session. It ignores the `expandIncludes` parameter.

With that said, either approach works for me. I am working on tests which cover the API for source symbols. Currently I've implemented the VB side the same way I implemented the C# side in this pull request. If you prefer it the other way, I can switch them all.